### DISCUSSION TITLE
Debug Setup on PUB400

### DISCUSSION THREAD

**Discussion by SJLennon** [codefori/vscode-ibmi#2036] [General]
I'm seeing this on PUB400:
![image](https://github.com/codefori/vscode-ibmi/assets/67484051/76c7b8e1-5761-4a08-aaff-7641174b98a4)

The file does exist, but it is from Apr 4, and is owned by someone else, so I can't delete it.  :
```
[-bash-5.2.15:certs]
=======>ls -al
total 28
drwxr-sr-x 2 train248 0  8192 Apr  4 17:19 .
drwxr-srwx 3 qsys     0 12288 May  2 14:30 ..
-r--r--r-- 1 train248 0  2499 Apr  4 17:19 debug_service.pfx
```
I suspect that the content is out of date since your recent changes.

Is the solution:

- Ask Holger to delete it an it will get recreated the next time someone tries debug? It does seem to get created with read privileges for everyone.

Or

- Ask Holger to delete it and then start debug in the extension so the `debug_service.pfx` gets created and owned by QSECOFR.?

**Comment by sebjulliand** [comment]
> I suspect that the content is out of date since your recent changes.
> 
It is indeed. The message should be `The certificate uses an old format and must be re-generated`, but you get the idea.
Ideally, Holger or an admin should:
- Connect
- Go to the debug view
- Generate the certificate from the `IBM i Debugger` view
- Start the Debug Service

After that, the Debug Service will run and it will be able to debug any program in any location.
Since the certificate will be generated using the new method, it will be possible for Holger to start the Debug Service after an IPL, using a simple `SBMJOB`, as described here: https://codefori.github.io/docs/developing/debug/#starting-the-debug-service-outside-of-code-for-ibm-i

@chrjorgensen since you are already in contact with Holger, can you ask him to do this?

**Comment by chrjorgensen** [comment]
> since you are already in contact with Holger, can you ask him to do this?

@sebjulliand Done! 😃

**Reply by sebjulliand** [reply]
You're the best my friend!

**Comment by mlstoppa** [comment]
Hi guys!
It looks like it's not working yet! I've just tried and it says certificates are not created and a user profile with *allobj authority is necessary to do so.
Do you know if and when both the certificate and the debug server will be available?
Thank you very much
Lucia

**Comment by chrjorgensen** [comment]
@mlstoppa Unfortunately the PUB400 administrator is very busy, so the debug _service_ is not yet available. I hope to have a face-to-face chat with him at the CEAC meeting in the end of this month, though.

We will post when the debug service is available. In the mean time you can still use the STRDBG command in 5250 and debug your programs in here.

**Reply by mlstoppa** [reply]
Ok, thank you.
Yes, I can keep on using 5250 strdbg command, but to me the point is to test the Code4i debugger, so I wait.
Thank you all

**Reply by mlstoppa** [reply]
Hi @chrjorgensen I talked to Holger at CEC24 in Milan for other stuff and asked even for debugging on PUB400: he said he will run the server in a week or so.
Cross fingers.

**Reply by Liscuri** [reply]
@mlstoppa good news! thanks!

**Reply by dcrdomains** [reply]
@chrjorgensen Any news on this? I have the same error message regarding certificates and *allobj authority.

**Reply by mlstoppa** [reply]
As regards this topic, I would share my thoughts: PUB400 is hosted by https://www.rzkh-gmbh.de, and they offer a developer account (https://www.rzkh-gmbh.de/developers/) for a reasonable fee. I have tried to reach them out since February, as I had considered the idea of paying to have a chance to test all of this stuff on a proper machine. No way, I got no answer at all. At this point, I think I will look for other paid solutions to be able to make my tries.

**Reply by chrjorgensen** [reply]
@mlstoppa I also talked to Holger at CEC2024 in Milan about this - and was told it would be fixed soon... but the issue has not yet been fixed, and there's no info of when that will happen. I'm sorry for this, as many users are asking for this - the only advice is to try to contact Holger themselves at [hs@rzkh.de](mailto:hs@rzkh.de), maybe they will have more luck. 🙏

**Comment by eulaubscher** [comment]
Morning all.  Any new news on this topic/discussion?  Thanks Eugéne.

**Reply by mlstoppa** [reply]
None at all!
